pymc3/distributions/continuous.py

```diff
@@ -1653,6 +1654,58 @@ def logcdf(self, value):
             tt.switch(tt.gt(y, 1), tt.log1p(-0.5 * tt.exp(-y)), tt.log(1 - 0.5 * tt.exp(-y))),
         )
 
+class AsymmetricLaplace(Continuous):
+    """
+    Assymetric Laplace Distribution
+    See also: https://en.wikipedia.org/wiki/Asymmetric_Laplace_distribution
+    """
+
+    def __init__(self, scale, symmetry, testval=0.0, *args, **kwargs):
+        """
+        Constructor
+        :param scale: scale parameter
+        :param symmetry: asymmetry parameter. Reduces to a normal laplace distribution with value 1
+        """
+        self.scale = tt.as_tensor_variable(scale)
+        self.symmetry = tt.as_tensor_variable(symmetry)
+
+        super().__init__(*args, **kwargs, testval=testval)
+
+    def random(self, point=None, size=None):
+        """
+        Draw random samples from this distribution, using the inverse CDF method.
+        :param point: not used
+        :param size: size of sample to draw
+        :return: Samples
+        """
+        if point is not None:
+            raise NotImplementedError('Random not implemented with point specified')
+
+        if size is not None:
+            u = np.random.uniform(size=size)
+            x = - tt.log((1 - u) * (1 + self.symmetry ** 2)) / (self.symmetry * self.scale) * (
```

Comment:
`self.symmetry` can be a random variable or just anything wrapped in [as_tensor_variable](https://theano-pymc.readthedocs.io/en/latest/library/tensor/basic.html#theano.tensor.as_tensor_variable). We certainly need to draw its values from model context.
You can do something like

```python
scale, symmetry = draw_values([self.scale, self.symmetry], point=point, size=size)
```
and then use its concrete values.